concordium-contracts-common/src/impls.rs

```diff
@@ -1251,6 +1251,20 @@ impl<T: AsRef<[u8]>> Read for Cursor<T> {
     }
 }
 
+// Unfortunately this has to be implemented for a mutable borrow version, since
```

Comment:
It is unfortunate, because it deviates from [the implementation in rust std](https://doc.rust-lang.org/std/io/struct.Chain.html#impl-Read-for-Chain%3CT,+U%3E).
I'll change the comment to include this information instead.